pkg/processor/build/util/decompress.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/nuclio/nuclio/pkg/errors"
+
+	"github.com/mholt/archiver"
+	"github.com/nuclio/nuclio-sdk"
+)
+
+type Decompressor struct {
+	logger nuclio.Logger
+}
+
+func NewDecompressor(parentLogger nuclio.Logger) *Decompressor {
+	newDecompressor := &Decompressor{
+		logger: parentLogger,
+	}
+
+	return newDecompressor
+}
+
+func (d *Decompressor) Decompress(source string, target string) error {
+	fileArchiver := archiver.MatchingFormat(source)
+	if fileArchiver == nil {
+		return fmt.Errorf("File %s is not compressed or has an unknown extension", source)
+	}
+
+	d.logger.DebugWith("File is compressed, now decompressing",
+		"file", source,
+		"compression", reflect.TypeOf(fileArchiver),
```

Comment:
Don't `fileArchiver`s have an accessible string representation method for their compression type?